src/sentry/profiles/consumers/process/factory.py

```diff
@@ -32,3 +38,12 @@ def create_with_partitions(
             health_checker=self.health_checker,
             next_step=next_step,
         )
+
+
+def get_headers_dict(headers) -> Dict[str, str]:
+    h = dict()
+    for k, v in headers:
```

Comment:
Are the headers a list of tuples or do you need `items()` here?

Reply:
yep, list of tuples: https://github.com/getsentry/arroyo/blob/0b4126f52509dc823e592d076747b34febcb3799/arroyo/backends/kafka/consumer.py#L63